### ISSUE TITLE
Random "The service is unavailable." and "Azure Functions runtime is unreachable" errors

### ISSUE THREAD

**Issue by Arjan321** [Azure/azure-functions-host#8583]
We have been running a couple of Azure Functions in various subscriptions, and every once in a while (about once a week), the entire Azure Function goes down and reports "The service is unavailable." when accessing the Function-app via HTTP and reports "Azure Functions runtime is unreachable" in the Azure Portal.

HTTP response:
![image](https://user-images.githubusercontent.com/47455366/181205074-77b2137e-c0b9-4a9d-a5d4-d45674621435.png)

Portal:
![image](https://user-images.githubusercontent.com/47455366/181205101-35d00594-8eac-4046-a724-7110bd93f861.png)

The issue appears randomly, without any changes in our end (no deployment, etc.) and also resolves randomly without any interaction from our side.

In the "Activity log" an error is written for the job  "Sync Web Apps Function Triggers" with status "Failed":
```
        "statusCode": "BadRequest",
        "statusMessage": "{\"Code\":\"BadRequest\",\"Message\":\"Encountered an error (ServiceUnavailable) from host runtime.\",\"Target\":null,\"Details\":[{\"Message\":\"Encountered an error (ServiceUnavailable) from host runtime.\"},{\"Code\":\"BadRequest\"},{\"ErrorEntity\":{\"Code\":\"BadRequest\",\"Message\":\"Encountered an error (ServiceUnavailable) from host runtime.\"}}],\"Innererror\":null}",
```

The issue seems quite similar to #8519, however we are running Linux.

This is causing quite some problems, since we are no longer able to provide reliable service to our end-users. 

#### Investigative information
- Timestamp: Wednesday, 27 July 2022, 07:47, Coordinated Universal Time (UTC) 
- Function App version: V4
- Function App name: Please use Invocation ID
- Function name(s) (as appropriate): All functions within the app are non-working
- Invocation ID: None of the functions work, but to find our function-app: 2022-07-27T08:05:13.773Z - e95ba16f-f2a2-420c-bcfd-4891d54198ae
- Region: West-Europe

#### Repro steps
None that we can find

#### Expected behavior
Always work

#### Known workarounds
None

#### Related information 

Hosting Model: Consumption Plan
OS: Linux
Version: V4
Hosting Model In-Process
Language: C#/dotnet6
Configuration:
 - WEBSITE_RUN_FROM_PACKAGE points to a valid file on a storage account
 - WEBSITE_MOUNT_ENABLED: 1
 - WEBSITE_ENABLE_SYNC_UPDATE_SITE: false
 - WEBSITE_CONTENTAZUREFILECONNECTIONSTRING: Valid keyvault reference toa a storage account
 - SCM_DO_BUILD_DURING_DEPLOYMENT: false
 - FUNCTIONS_WORKER_RUNTIME: dotnet-isolated
 - FUNCTIONS_EXTENSION_VERSION: ~4
 - AzureWebJobsDisableHomepage: true

**Comment by wbail** [comment]
Hi Arjan32,

Normally this error message is related with misconfiguration.

Please take a look the bullets below:

- Add the variable `AzureWebJobsStorage` and add the value of the connection string from the Storage Account
Reference:
  - https://docs.microsoft.com/en-us/azure/azure-functions/functions-app-settings#azurewebjobsstorage

- Double check the value of the variable `WEBSITE_RUN_FROM_PACKAGE`
Reference:
  - https://docs.microsoft.com/en-us/azure/azure-functions/run-functions-from-deployment-package 
  - https://github.com/projectkudu/kudu/wiki/WEBSITE_RUN_FROM_PACKAGE-and-WEBSITE_CONTENTAZUREFILECONNECTIONSTRING-Best-Practices

**Comment by Arjan321** [comment]
As mentioned in then original issue, Those values point to valid values. 

Given the extreme randomness of both the issue appearing and resolving itself, I hardly doubt any setting on our end is responsible.

**Comment by gilesmatthews** [comment]
Hi, is there any update on this issue? I am seeing the same random service unavailable issues. Thanks

**Comment by rlucassen** [comment]
Same issue here, completely random. Waiting on a fix for a while now.

**Comment by LuckyLub** [comment]
Sorry for you guys, but I'm happy to read this... thought I was going crazy. What region do you guys have your Azure Functions running? Ours are in West Europe.

**Comment by rlucassen** [comment]
> Sorry for you guys, but I'm happy to read this... thought I was going crazy. What region do you guys have your Azure Functions running? Ours are in West Europe.

Mine are running in West Europe as well, same for @Arjan321

**Comment by tomabg** [comment]
same region West Europe

**Comment by tomabg** [comment]
this also breaks terraform deployment...as we are on test only we will try different region soon

**Comment by rlucassen** [comment]
> this also breaks terraform deployment...as we are on test only we will try different region soon

Curious to know if other regions do work.

Heard that Microsoft is doing updates in the West Europe region around next week

**Comment by LuckyLub** [comment]
We also deploy with Terraform BTW. Changing the location to Central US seems to do the trick.

**Comment by jnekrasov** [comment]
Any news on this?! We are experiencing the same problems in West Europe region

**Comment by Ralle1986** [comment]
Also seeing issue in West Europe region

**Comment by balag0** [comment]
Sorry for the delay in responding. Yes, there was a regression due to a recent update in the region which caused intermittent errors. 
The fix rollout has already started and in progress currently. If there are any apps still experiencing failures, could you please share the details and we can double check them. Thanks

**Comment by LuckyLub** [comment]
Before we move back, can you please confirm when the roll-out is completed?

**Comment by lightwaver** [comment]
any news on that topic or link to the issue to see if its solved ?

**Comment by LuckyLub** [comment]
@balag0 any update?

**Comment by LuckyLub** [comment]
@surgupta-msft, @balag0, any updates? Just tried to redeploy to West-Europe, still running into random "The service is unavailable." errors Running the same functions in North-Europe works just fine.

**Comment by LuckyLub** [comment]
Just had contact with Azure's Help + Support. I presented the problem, it seemed to be known. However, they will collect some logs from my Azure Functions and look further into it. They are still working on upgrading the services in West-Europe. Currently it was advised to use another region.

**Comment by jeroenvermunt** [comment]
I can confirm that it is still occurring

**Comment by Rutix** [comment]
We have had these problems too. We have been in contact with Azure Support and they are saying the following:

"
**Conclusion**:
The 503s were detected ONLY on Azure Front End, the Front End instances encountered some unexpected error at that moment and weren’t able to handle the http requests and distributed them to specific workers that were hosting your function app.

..... <information containing sensitive data>

Let’s take a closer look at these 2 Front End instances at that time, Front End instance 24 encountered an error when trying to get a worker from the data role, and the same situation for the instance 5.


![image](https://user-images.githubusercontent.com/241452/196187558-7f525287-f7ec-489b-a2b2-7ed99dd879e1.png)
![image](https://user-images.githubusercontent.com/241452/196187586-4764304c-fc07-4e4c-880e-57c9f4a30444.png)

Unfortunately, this is an underlying platform issue as the Front End is an important component inside of Azure platform, and both the user side and our side cannot have action to interact with it, we apologize for all the inconvenience caused, but please rest assured that I’ve already reflected this to the Microsoft Azure Team and they already know this, we met this kind of issue before.
"

^ this has been now several months ago. The fact that these problems are still popping up is disappointing. Also telling us to a different region is also disappointing. We are bound by compliance issues so we cant leave our region as easy.

**Comment by rlucassen** [comment]
Last wednesday we got the message from Microsoft that this issue should be fixed, we already transferred to a premium plan because we got tired of it after 4 months. I'm curious to know if anybody is still encountering this issue in the West-Europe region?

**Comment by LuckyLub** [comment]
So using premium is a valid work around?

**Comment by rlucassen** [comment]
> So using premium is a valid work around?

Yes upgrading to premium worked for us

**Comment by Arjan321** [comment]
> So using premium is a valid work around?

If "Throwing money at the problem" counts as a workaround, then yes. This ticket is specifically about the Consumption Plan.

**Comment by LuckyLub** [comment]
Just good to know what options are out there.

**Comment by LuckyLub** [comment]
Message from MS:

Product Group is still working on improvement on the west Europe region. Will keep you updated on any progress.

**Comment by Rutix** [comment]
I got some messages that we got hit again. We are not entirely sure if it was the same cause as this issue but we had "Service unavailable" a couple of times today in west europe.

**Comment by rdvansloten** [comment]
@Rutix I happened upon this thread as well this afternoon, also from The Netherlands, having Functions (Consumption tier) in West Europe. They were unavailable and/or throwing SSL errors. I also could not deploy from VS Code (unavailable)

What fixed it for me is going into the Portal and Restarting my Functions manually.

**Comment by LuckyLub** [comment]
That’s a temporary fix.

**Comment by Rutix** [comment]
@rdvansloten sadly that is only a temp fix and also doesn't always work (or takes a long time to recover). The best way would be for the Azure team to fix this asap.

**Comment by rdvansloten** [comment]
@Rutix yeah it is, but it's better than "nothing" if you're using this in Prod. It keeps happening for me and a reboot is a fast fix for now. Did you put in a support ticket?

I get this diagnostic when creating a ticket:

```

Description | Function was running on 0 worker instance for more than 825 minutes between 10/26/2022 9:20:00 PM and 10/27/2022 11:10:00 AM.
-- | --
Possible Cause: | Function App was offline due to previous deploymentPlease restart the Function site manually or redeploy the function site to get the issue mitigated or use the AppOffline History detector to check if the function was offline during this period. Please visit Azure App Service Deploy task for more information. Function app site was Stopped or disabledPlease use the Web App Restarted detector to check if the function was stopped or disabled during this period.

</div></div>Host Runtime instance (Dynamic Plan) was not available for a long time period (> 15 minutes)
Description	
Function was running on 0 worker instance for more than 825 minutes between 10/26/2022 9:20:00 PM and 10/27/2022 11:10:00 AM.

Possible Cause:	
Function App was offline due to previous deployment

Please restart the Function site manually or redeploy the function site to get the issue mitigated or use the [AppOffline History ](https://applens.azurewebsites.net/subscriptions/865f86e6-0a9a-4c2f-8742-ce207e509dad/resourceGroups/myRG/sites/myApp/detectors/FunctionAppOfflineHistory?startTime=2022-10-26T11:25&endTime=2022-10-27T11:10)detector to check if the function was offline during this period. 

Please visit [Azure App Service Deploy task](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-rm-web-app-deployment?view=azure-devops) for more information. 

Function app site was Stopped or disabled

Please use the [Web App Restarted ](https://applens.azurewebsites.net/subscriptions/865f86e6-0a9a-4c2f-8742-ce207e509dad/resourceGroups/myRG/myApp/detectors/webapprestart?startTime=2022-10-26T11:25&endTime=2022-10-27T11:10)detector to check if the function was stopped or disabled during this period.
```

Apparently it was running for 825 minutes? Odd.

**Comment by Rutix** [comment]
> @Rutix yeah it is, but it's better than "nothing" if you're using this in Prod. It keeps happening for me and a reboot is a fast fix for now. Did you put in a support ticket?

Indeed it's better than nothing :) but would be better if Microsoft just fixes the problem. And yea we have had support tickets open. You can see the answer in one of my earlier comments. They know it's a known issue

**Comment by LuckyLub** [comment]
Another update from MS, they claim it has been fixed!

_The Microsoft Azure Team has investigated the issue you reported on your devOps pipeline that resulted in errors of your consumption apps becoming unavailable. This issue was found to be related to an issue within the capacity restriction in the region._

_Your consumption app was placed in a region with not enough capacity to fulfill all of the traffic. Because the West Europe region has one of the highest traffic for our product, we had to request an increase in the capacity through our partner team. After increasing the capacity for the region, the region was able to handle the high traffic and the availability issue was resolved._

_We are continuously taking steps to improve the Azure Web App service and our processes to ensure such incidents do not occur in the future, and in this case it includes (but is not limited to) ensuring high availability throughout the globe._

_We apologize for any inconvenience._

**Comment by stooone** [comment]
Happened with me yesterday and today too.

**Comment by rdvansloten** [comment]
We moved our Functions to North Europe. EUW is a damned mess right now. I couldn't deploy a new Function in EUW because capacity was mysteriously gone.

**Comment by hegde89** [comment]
we are also experiencing the same with EUW.  is there any update

**Comment by Rutix** [comment]
@rdvansloten been a couple of days, has North Europe treated you better? If so we will also make the efforts to move to North Europe

**Comment by jeroenvermunt** [comment]
@Rutix Our function runs at low volume (10 times a day), but since we moved to NE a few weeks ago I have yet to run into this issue

**Comment by rdvansloten** [comment]
> @rdvansloten been a couple of days, has North Europe treated you better? If so we will also make the efforts to move to North Europe

Not a single hitch since I put everything in North Europe.

**Comment by Rutix** [comment]
We have experienced the problem again in West Europe. We bit the bullet and moved the function to North Europe and that seems to help for now. So really seems to be a region issue like suggested earlier in the thread already. We made an Azure Support Issue again because its becoming kinda ridiculous how long this is taking Microsoft to solve.

**Comment by akselikap** [comment]
Also banging my head against the wall with this issue.

**Comment by stevengaaa** [comment]
We have experienced the same problem in Australia East today for 4hours starting from 4am (utc time) . Linux, Consumption tier. but Windows based functions are running ok. apps on Premium plan are ok, too.

**Comment by Rutix** [comment]
We contacted Azure Support and this is the message they gave back:

> Upon further checking, there is an emerging issue ongoing which is impacting a lot of Function apps on Linux Consumption plan in West Europe region.
Unfortunately, the stamp that your app is hosted on is impacted.
> 
> Our Product Group team is actively working on this issue right now and once there is any further update, I will let you know as soon as possible.
>  
> Please accept our sincere apologies on the inconvenience this issue has caused. Typically, when similar issue occurs, as we have backend algorithm to detect and auto heal such issues, your app should be recovered within 0.5 hour. That is why normally issue like this should be a "one-time issue".
> 
> However, this time it is an emergent issue on the platform side which cannot be mitigated by our backend automatically, and the corresponding team is needed to be engaged to fix the issue. We will keep you posted over the progress.
> 
> If the issue is business critical and you will need the app to be working immediately, we will recommend temporarily moving to a dedicated plan as the apps hosted in dedicated plans are using a different set of stamps (worker machines) and are not impacted by this issue.
> 
> Hope the above could be helpful to you. If you have any further question or concern, please feel free to let me know anytime. We are always here to help.

**Comment by bcdunbar** [comment]
I haven't read through all the above but dropping a note here in case it is useful for others.

We experienced this random error yesterday (Australia Southeast), which the runtime was unreachable, but Function App was running and stating healthy. We checked storage accounts, rotated keys, restarted app, but none worked. In the end, we migrated v3 to v4 locally, confirmed it was working, and then used forced deployment to resolve the issue (as regular deployment was failing). 

Migration: https://learn.microsoft.com/en-us/azure/azure-functions/migrate-version-3-version-4?tabs=net6-in-proc%2Cazure-cli%2Cwindows&pivots=programming-language-python 

Force deployment:

```
func azure functionapp publish <FUNCTION-APP-NAME> --force --verbose
```

Perhaps the migration is not needed, and a forced redeployment may resolve the issue. A caveat here is that this is likely a temporary fix but there is no indication as to what the root cause was given there had been no changes to that service for a month.

**Comment by Rutix** [comment]
We received another update from Azure support yesterday:

> Summary of Impact: Between 01:19 UTC and 10:31 UTC on 14 Dec 2022, you were identified as a customer using App Service in West Europe who may have received intermittent > HTTP 500-level response codes while creating the resource, experienced timeouts or high latency when accessing App Service (Web, Mobile, and API Apps), App Service > (Linux), or Function deployments hosted in this region.
> 
>  
> 
> Preliminary Root Cause: We determined a backend service experienced failures due to a disabled component resulting in the above failures.
> 
>  
> 
> Mitigation: We manually enabled the component of the backend service to mitigate the issue.
> 
>  
> 
> Next Steps: We will continue to investigate to establish the full root cause and prevent future occurrences. Stay informed about Azure service issues by creating custom > service health alerts: https://aka.ms/ash-videos for video tutorials and https://aka.ms/ash-alerts for how-to documentation. 
> 
>  
> 
> Once a full root cause analysis is available, I will let you know as soon as possible. Thank you again for your patience and understanding.

**Comment by apoorvmintri** [comment]
I've just started to experience this in **East US**; unable to resolve for the past 2 hours. Any help would be really appreciated!

Edit: Forgot to mention that this was on P1 Plan (.NET 6, Linux, Code deployment i.e. no docker)

Update: Went for lunch, came back and it works now - no changes made. Is Funtion v4 production ready?

**Comment by vgool** [comment]
Good to hear I'm not the only one suffering this issue..

My Functions (windows, consumption plan, West Europe) deployed to our Development environment run very slow and fail randomly, however the Functions on our Test environment run fast and all succeed..

Unbelievable this issue is open for over half a year!

**Comment by markleavesley** [comment]
We are also experiencing intermittent "HTTP Error 503. The service is unavailable" for v1 Functions (Windows, consumption) in UK South and Australia Southeast, more than 40 occurences in the last week. They are v1 because we're stuck on Framework SDKs for these two and we are waiting for the v4 Framework tooling to settle down and give that a go (last time I tried it the template HTTP function wouldn't even compile!)

**Comment by bdorplatt** [comment]
We are seeing this exact issue as well in the North Central US region.

**Comment by markleavesley** [comment]
I raised this with MS who basically said upgrading to v4 Isolated Functions fixes it, which is something we were already looking into anyway as v1 is no longer being maintained. We didn't feel inclined to push for any actual explanation as to what was going on as the v4 upgrade seems fairly quick and painless tbh.

https://learn.microsoft.com/en-us/azure/azure-functions/migrate-version-1-version-4?tabs=v4%2Cazure-cli%2Cwindows&pivots=programming-language-csharp

I found creating a v4 Isolated Function (http trigger, Framework 4.8) was a useful support to the instructions as they seem to be slightly out of date.

**Comment by Alebat** [comment]
Still same issue here for WEu region

**Comment by bdorplatt** [comment]
After troubleshooting, we found that these issues, in our case, were due to Host ID collisions as detailed in the article below. Specifically this:
 "When you have function apps with names longer than 32 characters and when the first 32 characters are identical, this truncation can result in duplicate host ID values. When two function apps with identical host IDs use the same storage account, you get a host ID collision because stored data can't be uniquely linked to the correct function app."

Our naming standard for certain applications caused some apps to have the same 32 character Host ID. Once we shortened the names to accommodate for this, we no longer saw the collisions or the "The service is unavailable." and "Azure Functions runtime is unreachable" errors.

https://learn.microsoft.com/en-us/azure/azure-functions/storage-considerations?tabs=azure-cli#host-id-considerations

**Comment by rdvansloten** [comment]
@bdorplatt I am seeing this issue with Fn Apps with names shorter than 20 characters that only share the first 6 characters in the name.

I also feel that this 32 character limit producing duplicates is the provider pushing a problem onto the user. The API should not allow us to create apps with "conflicting" names. Nobody is gonna find this article until it's "too late." 

Furthermore, renaming the apps may cause downtime or issues for those of us that bound DNS aliases to the apps or refer to their DNS name in other applications.

**Comment by gwsampso** [comment]
encountering this right now - out of nowhere stopped working

**Comment by ChandimaSharpCloud** [comment]
Experiencing the same. Suddenly all envionment stopped working

**Comment by RaulLpg** [comment]
Same issue here, seems to affect only West Europe but it is very scary that our apps would just randomly become unavailable and we can't do anything about it

**Comment by hristochr** [comment]
The same over at my end too, happened an hour ago and still ongoing. 

Update 1PM CET: issue gone
Update 2PM CET: issue reappeared, however, in the mean time I updated my production function to S1 from B1 and it's ok.

**Comment by kdlangekaak** [comment]
We also experience this problem. Also in the last hours in west Europe. Out of the blue unreachable without access to the logs, now it works again.

**Comment by KingofGnome** [comment]
Same here, West-Europe, `The service is unavailable.` on multiple functions

Redeployment using DevOps-Pipelines not possible, as the Service is Unavailable and therefore unable to Sync Triggers :/
`##[error]Error: Failed to sync triggers for function app '...'. Error: BadRequest - Encountered an error (ServiceUnavailable) from host runtime. (CODE: 400)`

**Comment by greigs** [comment]
We appear to be experiencing the same issue. All endpoints returning 503 and we are unable to deploy - same 503 error. Also West Europe.

**Comment by Rasmnev** [comment]
Same here. All apps stopped working. Also west europe

**Comment by pmooij** [comment]
Same here for Linux apps on consumption plan in West Europe.
(Win apps on consumption plan seem to be working though)

Just tweeted @AzureFunctions as well; https://twitter.com/paulmodus/status/1633449076577411073

**Comment by xBudii** [comment]
Same here, "The service is unavailable." - EUW and Linux App

**Comment by Erygh94** [comment]
Same here - Linux on consumption plan in West Europe

**Comment by JelleSmet-TomTom** [comment]
Same here - Linux on consumption plan in West Europe

**Comment by akselikap** [comment]
Is Azure violating the SLA with these interruptions in service? I have no clue what the difference between their app service plan and consumption plan SLA is but [the wording is definitely different](https://www.azure.cn/en-us/support/sla/functions/):

> For Function Apps running on Consumption Plans we guarantee that the associated functions will run 99.95% of the time after trigger is successfully fired.
> 
> For Function Apps running on App Service Plans we guarantee that the associated Functions compute will be available 99.95% of the time.

**Comment by Erygh94** [comment]
> Is Azure violating the SLA with these interruptions in service? I have no clue what the difference between their app service plan and consumption plan SLA is but [the wording is definitely different](https://www.azure.cn/en-us/support/sla/functions/):
> 
> > For Function Apps running on Consumption Plans we guarantee that the associated functions will run 99.95% of the time after trigger is successfully fired.
> > For Function Apps running on App Service Plans we guarantee that the associated Functions compute will be available 99.95% of the time.

Good question. They probably are. I also find it a bit troublesome that all services are still in the green on Azure status almost 7 hours after I received my first errors (08:12 AM UTC+1)

**Comment by Erygh94** [comment]
Appears to be working again now as of 16:17 UTC + 1 without any changes on our end.

**Comment by KingofGnome** [comment]
> Appears to be working again now as of 16:17 UTC + 1 without any changes on our end.

Same here, seems to work again

> Good question. They probably are. **I also find it a bit troublesome that all services are still in the green on Azure status** almost 7 hours after I received my first errors (08:12 AM UTC+1)

Absolutely agree there!
Trying to open up a ticket from the Function-Resource directly we've seen an Impack Statement, mentioning something like
> An Ayure service issue impacted resources in your subscription ....using App Service in West Europe may receive HTTP 500 responses, timeouts or high latency...

After investigating multiple hours into failures from our side :/

**Comment by DavidAtImpactCubed** [comment]
To add to the list of impacted users, we are seeing the same issue with Linux consumption plans in West Europe. Our production Linux function app is still running (Linux/WE) but our pre-production environments, which are deployed to more frequently, have fallen over.

**Comment by Erygh94** [comment]
We experienced this again today with first error at 7:29 am UTC+1 and first working response again at 8:07 am on Azure functions running ubuntu-latest on consumption plan EUW.

**Comment by DavidAtImpactCubed** [comment]
We had the same, for 45 minutes this morning. Same host as last time.

**Comment by Erygh94** [comment]
Service is unavailable again as of 12:22 PM.

**Comment by DavidAtImpactCubed** [comment]
Same for us, as of 12:10 UTC

**Comment by espenja** [comment]
Also experiencing this right now deploying from Azure DevOps. ARM-templates ran just fine. Deployed to staging environment and the deploy fails with error: `##[error]Error: Failed to sync triggers for function app 'name-APP-prod-staging'. Error: BadRequest - Encountered an error (ServiceUnavailable) from host runtime. (CODE: 400)`

After checking the Azure Function App, all our Functions have disappeared in both slots and we are seeing error "Azure Functions runtime is unreachable" under the "Functions tab". All our post processing is now down in production, once again.

This is the umpteenth time we're facing this issue. Starting to get real fed up. Completely unreliable when it fails or not.
We have made no changes to setup or configuration of the Azure Functions, only code changes.

**Comment by DavidAtImpactCubed** [comment]
Presumably this your app is also in West Europe on a Linux consumption plan @espenja?

What I find most troublesome of all of this is that the [Azure status page](https://azure.status.microsoft/en-gb/status) has had a green tick against Azure Functions in West Europe throughout, even whilst the Azure support team is [tweeting](https://mobile.twitter.com/AzureSupport/status/1633512227113422850?cxt=HHwWhMDSleits6stAAAA) that they are investigating an issue. 

It just seems a bit disingenuous. If you know there is a degradation of service, and you are investigating, own up to it on your status page (and honour your SLAs).

**Comment by espenja** [comment]
@DavidAtImpactCubed That's correct. We're running Linux Consumption plan in West Europe. We have been in contact with Azure Support with grade A errors numerous times about this exact same issue, and it keeps coming back to haunt us. The Azure Status page at this point is of no value to us. Last time we tried to report an error, everything was green on the status page, but we weren't allowed to actually report the error because "An internal investigation is ongoing" for the resource and resource type I was trying to report. Last week's failure where West-Europe failed completely was resolved already before the status page changed to warning signals.

**Comment by oyvind3** [comment]
I have three different function app running in West Europe, all three are reporting "The service is unavailable". Been running the same function app for almost three months without experiencing any troubles. My rational thinking is that "serverless" is just as server and infrastructure dependendt than the alternatives.

**Comment by DavidAtImpactCubed** [comment]
More outages today, this time in our production environment. 

There is a more recently opened issue on the static web apps repo where more people are reporting the issue: https://github.com/Azure/static-web-apps/issues/1097

**Comment by espenja** [comment]
We are having the same issues today. This time it happened without us doing anything. No release from DevOps, it just stopped working suddenly. I'm glad it's not only us reporting errors here, because we have no idea knowing if this is affecting anyone else when the Azure Status page is showing all green.

**Comment by arpanbanerjee** [comment]
We are also having this issue. 
Our ADF pipeline is calling an Azure Durable Function to move some files from Azure Blob to AWS S3. It ran successfully 3 times yesterday, however, suddenly we have 2 failures with this error message. 
No changes were done to the code, and as the function itself was not invoked, we did not find any logs on them.

**Comment by culpgrant** [comment]
Having an issue on this today

**Comment by carmunsRE** [comment]
I'm also experiencing this issue today since around 4 or 5 hours ago with a function that runs every 5 minutes.

**Comment by culpgrant** [comment]
Are you running on east?

**Comment by carmunsRE** [comment]
Yep, East US

**Comment by dsm320** [comment]
I am also currently facing this issue in the East US. How long does it typically take to randomly resolve itself?

**Comment by carmunsRE** [comment]
No idea, I'm creating a support request and got this message that the issues is tracked, but it says it's resolved and I'm still getting the error:

Resolved: An Azure service issue (Tracking ID _K5G-F88) impacted resources in your subscription.
Summary of Impact: Between 10:57 UTC and 11:32 UTC on 28 Mar 2023, you were identified as a customer using Azure Functions applications hosted on Linux Consumption SKU in East US who may have received intermittent HTTP 500-level response codes, experienced timeouts or high latency. Apps with running instances would have experienced throttled scaling.

**Comment by bartvanmiegem** [comment]
We are facing the same issue as described above. It's occurring time to time. Eventually it is available again, but we are bound to SLA's. Any suggestions/alternatives to improve the availability? We are doubting to spin up a dedicated instance to serve the api endpoint, but preferrably we are still using the Azure functions (if they are stable).

**Comment by carmunsRE** [comment]
One of the persons that had the issue months ago in West Europe said that the issue was not affecting premium plan, so you can try to upgrade.

**Comment by Rutix** [comment]
> We are facing the same issue as described above. It's occurring time to time. Eventually it is available again, but we are bound to SLA's. Any suggestions/alternatives to improve the availability? We are doubting to spin up a dedicated instance to serve the api endpoint, but preferrably we are still using the Azure functions (if they are stable).

The issue mostly happens in regions which are having capacity issues like west Europe. We moved to North Europe and haven't had problems since. Or you could move to premium plan if you are willing to pay.

**Comment by rfabro** [comment]
Also happens in my function running Consumption plan, located in Southeast Asia.

**Comment by potatoleaf** [comment]
Same happend to me with a function app running in west europe (also on consumption plan). Function app wasn't running for about 15 hours.

**Comment by Thue77** [comment]
We are experiencing the same thing running function on a Consumption plan in West Europe

**Comment by ccvaillant1992** [comment]
We are having the same issue with the azure function resource python runtime in the West Europe location
![image](https://user-images.githubusercontent.com/56896573/235881530-31a112cf-08ab-490e-a453-051ac483d0cc.png)

**Comment by greigs** [comment]
Yes this is also happening *yet again* for us - West Europe, consumption plan, linux. This service claims to be a SLA of 99.95% uptime, but we are getting nowhere close to that over the last few weeks.

**Comment by Thue77** [comment]
Seems like they are working on an issue, but once again, it does not show up in the health status..

![image](https://user-images.githubusercontent.com/43001325/235883585-b50dd856-adf5-447f-92e2-1659f71a11ba.png)

**Comment by SimonDania** [comment]
It's happening for me too, unfortunately.
Seems to be a bi-monthly occurance at this point. Really unfortunate, when it is affecting a production environment.

**Comment by Thue77** [comment]
I should be resolved now:

![image](https://user-images.githubusercontent.com/43001325/235902109-2ea898b9-3139-4f3c-bc5b-7bc8c3fe73ef.png)

**Comment by SimonDania** [comment]
It's working again in my end.
Had about 2-3 hours of downtime.

**Comment by Thue77** [comment]
Same

**Comment by greigs** [comment]
We had 4 hours downtime followed by 15 hours uptime and now things are down again and have been for 2 hours. This is not acceptable.

**Comment by fzoccara** [comment]
I had the same issue, on azure function linux based on consumption plan on west europe zone.
A request through postman get a "The service is unavailable."
We can't deploy too into that functions

**Comment by leidegre** [comment]
Had this issue myself, just now. resolved itself but this is worries some. We depend on this to work for critical infrastructure. Should we look elsewhere?

**Comment by flannoo** [comment]
Same issue happened for me (azure functions on linux consumption plan in west-europe). My azure functions went down a couple times today & yesterday with the error "The service is unavailable.". When checking my functions in the azure portal, I see the message "Azure Functions runtime is unreachable".

This seems to have happened a couple times for me in the past weeks :-(

**Comment by ottosulin** [comment]
This is ridiculous. Not only have we been experiencing the issue mentioned here, but also a bug causing the entire runtime to fail when running a custom handler in the Linux runtime - which Azure Support does not acknowledge as a bug, even though they can't point out what's wrong with our configuration.

The Functions dev team should really focus on reliability and observability of their service if they want to keep their customers.

We are planning to migrate to some other platform because it looks like Microsoft is just unable to keep this service running without significant issues and bugs all the time.

**Comment by giovannipapini-agilelab** [comment]
Same issues as those already reported here: unable to use (Error 503) and deploy to functions (on linux) with consumption plan.

**Comment by DavidAtImpactCubed** [comment]
We have redeployed our Linux function apps to North Europe (from West Europe) and touch wood, so far no issues. It does seem like this is a West Europe problem.

**Comment by slacker298** [comment]
currently experiencing this on US East - service randomly goes from running just fine to 503 "The service is unavailable."
was down most of this morning, then up for about 30minutes, then down another 1-2hrs, now up again.  Taking comfort in the fact that this is seemingly Microsoft's issue, as nothing has changed about my function.

Interesting side note, the times that it was down, I could get exactly one invocation of it after re-deploying before it went to responding with 503's only

Will also note that this is only an issue for my recently deployed Azure function (serverless python 3.10, deployed this past Friday).  All my other functions (also serverless 3.10, US East) have been just fine

**Comment by arpanbanerjee** [comment]
Is anybody running Durable Functions? We are not getting "Service Unavailable" there. 
Region: US East.

Edit:
Now the functions are up and running  (both 'normal' and Durable Functions). This is very random and unpredictable. Is there any way we can monitor when this might happen in a region, or, know why this is happening?

**Comment by ErikHen** [comment]
I have had the same 503-issue for more than a year now. Running Linux Function, Consumption plan, in West Europe. The issues comes and goes without any changes being made to config or code. Sometimes they last for minutes, sometimes the last for hours. 
A couple of weeks ago I moved to Windows-based Functions, and haven't experienced this issue after that.

**Comment by broesdecat** [comment]
I created some Azure Functions last week, Python on Linux, App Service plan, West Europe, and have not been able to actually use them, they have been down more than up...
Will try moving to Windows-based functions, thanks for the suggestion @ErikHen
Update: never mind, Basic ASP is 5x more expensive and no Python available on Windows...
Update 2: we noticed that our publicly-available Function App don't seem to suffer from this, but Function Apps deployed within our VNet (network injection) do.

**Comment by jrunestone** [comment]
I also have had this problem, linux host consumption plan west europe, my problem was related to enabling managed identity on the function app. Disabling that made it work again.

Edit: was trying to connect to cosmosdb via managed identity, but target service seems irrelevant since wp-sm below had the same error trying to access a different service via managed identity.

**Comment by ghost** [comment]
> I also have had this problem, linux host consumption plan west europe, my problem was related to enabling managed identity on the function app. Disabling that made it work again.

Thank you for mentioning this. For us this is the problem as well. 
Within seconds after creating a System Assigned Managed Identity on a Linux Consumption Plan function in West Europe, it crashes with the below error:
![image](https://github.com/Azure/azure-functions-host/assets/77006728/5efe43c7-9c1e-456d-b14f-665d4c7cf2aa)

The behavior is predictable and reproducable. 

Also potentially useful context info: We're trying to connect to an App Configuration instance via Managed Identity.

**Comment by mikkovor** [comment]
I thought it was a joke but yes if you enable managed identity on a linux consumption plan then you get this "Azure Functions runtime is unreachable" and if you disable it works again but we need the managed identity. What the hell?

**Comment by maaaNu** [comment]
Hi Guys,

Is somebody able to create a function app in euw? Neither with terraform nor manually I was able to create a function app for my service-plan. E.g. if I try to use the azure portal I am stuck in the "validating" step:

![Screenshot 2023-06-02 at 11 02 31](https://github.com/Azure/azure-functions-host/assets/5471583/b056d370-b814-4ae4-8207-43117bfac957)

**Comment by FFranz93** [comment]
Due to the quantity of comments I did not read all of them, but on our Function App (running on Windows with a Consumption Plan in West Europe) we are also facing the same issue randomly about once a week.

**Comment by milkyjoe90** [comment]
I've been experiencing this since 6pm last night, linux consumption app returning 503 for both http triggered calls and deployments from Azure DevOps pipelines and releases

**Comment by javast97** [comment]
If anyone is experiencing the same issue, I resolved refreshing the SAS Token.

See this discussion below:

#9113

**Comment by rdvansloten** [comment]
> If anyone is experiencing the same issue, I resolved refreshing the SAS Token.
> 
> See this discussion below:
> 
> #9113

I don't think this applies to most of us. This has been happening in West Europe with Functions that don't use SAS tokens or are less than a month old.

**Comment by javast97** [comment]
> > If anyone is experiencing the same issue, I resolved refreshing the SAS Token.
> > See this discussion below:
> > #9113
> 
> I don't think this applies to most of us. This has been happening in West Europe with Functions that don't use SAS tokens or are less than a month old.

Maybe not apply to all of you, but I'm not in this thread from a specific search "West Europe", I searched the error an this thread is in the top of searches and maybe helps anyone.

**Comment by AliVaseghnia** [comment]
The SAS token for the `WEBSITE_RUN_FROM_PACKAGE` url was expired in my case, updating the url fixed the issue for me.

**Comment by mrubiottec** [comment]
This has intermittently occurred on our end as well. Any update?

**Comment by ottosulin** [comment]
We also see these on our Static Web Apps regularly about weekly for short moments. Afaik SWA backend run on Function Apps.

**Comment by mrubiottec** [comment]
Microsoft has to act on this quickly. This looks to be a global problem.

**Comment by noontz** [comment]
I have the exact same problem deploying a completely new unaltered azure function template from Visual Studio to a North Europe App Service Plan. The pipeline succeeds with no issues or warnings..

**Comment by bluebobbo** [comment]
Yup, same exact issue here. East US. I also no longer see the "your function app 4.0" is running page when I go to the function's URL. Strangely enough, a few weeks ago this was all good.

I ended up deleting the function app and service plan, recreating it in, and then things worked normally for a couple hours.

Now back to the same issue.

**Comment by bluebobbo** [comment]
I tried deleting the function app and the app service plan. Then re-created the resources and now things are working like they should.

**Comment by isadag** [comment]
I thought I was going crazy until I found this post. I have a relatively new function that has now experienced this twice (today, and a couple of days ago). It is really frustrating that there is now workaround or fix. I haven't been able to find any status or outage info either when this occurs. The 503 Service Unavailable occurs without any changes in the code or settings and then resolves itself after quite some time (took about 1 hour ish). Restarting the function app has no effect whatsoever.

Running Linux Function, Consumption plan, in Sweden, and in this case I'm having issues with my http triggered function. I've tried West Europe too, but got the exact same issue there too..

Any news on this? Any tips to avoid this behavior? 

Isa

**Comment by isadag** [comment]
I wanted to add a follow-up comment since I've been comparing two different accounts (my own personal, and an account belonging to an organization I supported with some setup previously.) The thing I find interesting is that when I check both accounts and compare some stats, I see the same information for both accounts, which I assume to be because of a general (even global ish, or at least in the same region?) issue in Azure.

**What stats did I look up and how?**
1. In the Azure portal, go to your function app
2. In the left menu, click the link called **Diagnose and solve problems**
3. Click the search bar and wait a few seconds for it to load before you type in: **Function App Down or Reporting Errors**
4. Click on that item in the search bar's dropdown
5. In the time-picker you should look at the day you had issues, in my case I chose to set the time-picker to be on Wednesday the 17th of January (one of the days I had issues with my function apps), and the time is from 00:00 until the 18th at 00:00.
6. Let it load..... once it has analyzed the data you should get some information and stats
7. Scroll down to panel containing info about the issues, the panel has the heading: **Host Runtime instance (Dynamic Plan) was not available for a long time period (> 15 minutes)**


Both accounts (not connected to each-other in any way) are running a Linux Function, Consumption plan, in Sweden Central.

When observing the report in the Azure Portal I can see that the functions had issues for a total of **470 minutes** where the function was running on **0 worker instance**.. And on the 21st of January (Sunday) there seemed to be issues as well, but this time for a total time of **1435 minutes**. 

**I have not found any outage information from MS.**

**If you are running azure functions, could you take 5 minutes to:**
* look up those two dates (17/1 and 21/1) and get back here with info whether you had the same issue and what info is stated there? 
* write info about type of function, OS, which region you're running in and whether you're running consumption/premium/dedicated..

<img width="717" alt="image" src="https://github.com/Azure/azure-functions-host/assets/18397374/81d7be6e-13ba-4076-906b-025903685010">

**Comment by tom08zehn** [comment]
Thanks @isadag for this step-by-step guide.

I'm running an Azure Function with an HTTP trigger (Node.js 18) on a Consumption plan in France Central and I'm also running into random HTTP 500 errors. Workaround: refresh page. Impact: unstable service to users. It's definitely not an error caused by my app because I wrapped the entire code in a try-catch-block and Application Insights does not show any application error.

**Any ideas for a solution or further investigation?**

For me it reports 2 errors (full exception below):

1. Functions that are not triggering

```
Host Runtime instance (Dynamic Plan) was not available for a long time period (> 15 minutes)

Description: Function was running on 0 worker instance for more than 1420 minutes between 1/25/2024 8:20:00 AM and 1/26/2024 8:05:00 AM.

Possible Cause: Function App was offline due to previous deployment
```

![image](https://github.com/Azure/azure-functions-host/assets/12202052/6869ffba-a2f2-47ee-be45-f886b12db740)

2. Function Executions and Errors

`Detected function(s) having execution failure rate between 0.1% and 1%.`

![image](https://github.com/Azure/azure-functions-host/assets/12202052/b84e8616-34a9-4a63-8259-2c879dd969d2)

```
Timestamp : 1/25/2024 5:25:54 PM
Inner Exception Type: System.Exception
Total Occurrences: 1
Latest Exception Message:

Full Exception :
 Exception while executing function
 /Functions.register ---> Microsoft.Azure.WebJobs.Script.Workers.WorkerProcessExitException 
 /node exited with code -1073740791 (0xC0000409) ---> System.Exception
   End of inner exception
   
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   
async Microsoft.Azure.WebJobs.Script.Description.WorkerFunctionInvoker.InvokeCore(Object[] parameters,FunctionInvocationContext context) 
/_/src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs 
 /101
   
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   
async Microsoft.Azure.WebJobs.Script.Description.FunctionInvokerBase.Invoke(Object[] parameters) 
/_/src/WebJobs.Script/Description/FunctionInvokerBase.cs 
 /82
   
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   
async Microsoft.Azure.WebJobs.Script.Description.FunctionGenerator.Coerce[T](Task`1 src) 
/_/src/WebJobs.Script/Description/FunctionGenerator.cs 
 /225
   
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   
async Microsoft.Azure.WebJobs.Host.Executors.FunctionInvoker`2.InvokeAsync[TReflected,TReturnValue](Object instance,Object[] arguments) 
D:\a\_work\1\s\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionInvoker.cs 
 /52
   
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   
async Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.InvokeWithTimeoutAsync(IFunctionInvoker invoker,ParameterHelper parameterHelper,CancellationTokenSource timeoutTokenSource,CancellationTokenSource functionCancellationTokenSource,Boolean throwOnTimeout,TimeSpan timerInterval,IFunctionInstance instance) 
D:\a\_work\1\s\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs 
 /581
   
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   
async Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithWatchersAsync(IFunctionInstanceEx instance,ParameterHelper parameterHelper,ILogger logger,CancellationTokenSource functionCancellationTokenSource) 
D:\a\_work\1\s\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs 
 /527
   
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   
async Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithLoggingAsync(IFunctionInstanceEx instance,FunctionStartedMessage message,FunctionInstanceLogEntry instanceLogEntry,ParameterHelper parameterHelper,ILogger logger,CancellationToken cancellationToken) 
D:\a\_work\1\s\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs 
 /306
   End of inner exception
   
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   
async Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithLoggingAsync(IFunctionInstanceEx instance,FunctionStartedMessage message,FunctionInstanceLogEntry instanceLogEntry,ParameterHelper parameterHelper,ILogger logger,CancellationToken cancellationToken) 
D:\a\_work\1\s\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs 
 /352
   
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   
async Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.TryExecuteAsync(IFunctionInstance functionInstance,CancellationToken cancellationToken) 
D:\a\_work\1\s\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs 
 /108
```

**Comment by benrutter** [comment]
Seem to be having this same issue now (also in West Europe region). We've noticed random "service is unavailable" failures for a while - although recently an entire functions service was done for 1420 minutes between 1/29/2024 9:40:00 AM and 1/30/2024 9:25:00 AM.

No outage information reported from my end - and seems to have "fixed itself". As someone trying to debug this, having an entire service just stop working without any changes on our side is kind of crazy.

**Comment by tom08zehn** [comment]
For me the "service is unavailable" error (HTTP 500 status code) occurs in ~one out of 100 HTTP calls and it's immediately fixed when you reload the page (I'm serving a tiny website). There is no long outage but I see in the logs that the reason is a `System.Exception`.

**Does anybody know how to troubleshoot this with Microsoft?** Even if it's a consumption plan it's still a paid service.

**Comment by isadag** [comment]
I am wondering the same thing, how can this be reported and escalated to an MS team?? 

I find it very strange that there is no information on the status page regarding these outages that occur. The comments in this GitHub issue all show that it is not something that is application specific or as an effect of some action taken by us developers or ops team, but rather something in the functions platform that seem to be region(s) wide. It is very concerning indeed.

**Comment by noontz** [comment]
> **Does anybody know how to troubleshoot this with Microsoft?** Even if it's a consumption plan it's still a paid service.

Take the lack of feedback from MS on this issue as a clue.
From my own experience this will require a PAYED support subscription, and even then there will be a reluctance from MS to recognize / admit bugs on Azure. I had another case that took 2 months before the issue was even recognized, yet alone the time to find a workaround. And yes. I had to pay for that experience with a support subscription.

**Comment by tom08zehn** [comment]
I'm in contact with MS Azure support (my org has a paid support subscription)... will keep you posted...

**Comment by peterboba** [comment]
We're also in contact with MS support since we're experiencing HTTP 503/502 responses when trying to access
https://{func}.scm.azurewebsites.net/. Mostly WestEurope, but seen also in other regions.

**Comment by agravityio** [comment]
We are facing the same issue. The 502/503 occurs when we apply the User Access Managed Identity on the functions on EP1 plan. We always used the same bicep file, so it must be something on the side of microsoft. Tried to create function manually in the portal in West Europe at it succeeded. But we need to access the keyvault. So after applying the UAMI the 502/503 came back. 

I am running a Windows function app (.NET 6.0 in-process) on West Europe on Elastic Premium 1. So the "pay your way out" is not an option, since we are already paying.

@fabiocav - Do you have any thoughts on that?

**Comment by ajuch** [comment]
We're also affected by random 503 errors in our function app. Also in West Europe. We updated Microsoft.Azure.Functions.Worker.Sdk from 1.15.1 to 1.16.4 and updated .Net from 7 to 8. The function app works, but tests often fail with status code 503. This didn't happen before.

**Comment by agravityio** [comment]
Okay. I invested some time on this issue and I have to revert my mentioned thesis. In my case it does not have anything to do with UAMI.

Just for all others who run in to this issue. Until now it was not problem to have the following app settings enabled on creation:
`{
  name: 'WEBSITE_RUN_FROM_PACKAGE'
  value: '1'
 }
 {
 name: 'WEBSITE_ENABLE_SYNC_UPDATE_SITE'
 value: 'true'
 }`
After removing this the creation of the Function App was possible again.
They will be added after pipeline deployment anyway, but on creation this caused the 502/503 response status.

**Comment by isadag** [comment]
@peterboba and @tom08zehn did you guys hear anything back from MS Support regarding this issue? 🙏🏻

**Comment by tom08zehn** [comment]
MS is still investigating... and no news since one week. In my case it seems like the random HTTP 500 error occurs when my Node.js Function establishes a connection to the Azure SQL database (serverless) via the mssql module - the strange thing in my eyes is that my code retries all db activities (connect, query, etc.) but all this is not executed because the underlying Node.js process crashes... So it's either a general Node.js issue or a specific problem with the mssql module or with the Azure SQL db or with the Azure network or ...

**Comment by isadag** [comment]
OK, sounds strange. In my case the request never even hits my code, it failed as soon as the function app was being requested. Instead of getting the "your app is up and running" screen, I would get the 503 instead. 

For instance let's say I have an http endpoint here https://myfunction.azurewebsites.net/api/myendpoint. Even if I just tried to browse to the function app at: https://myfunction.azurewebsites.net I'd get a 503 instead of the "up and running" screen, and none of my code is being executed from there.

**Comment by tom08zehn** [comment]
In my case the Function code is executed. I see it because I added some verbose logging at the very beginning and before/after database queries.

At the moment, my gut feeling tells me that it's a general issue in the way Azure serverless works (trying to allocate resources and running the process) or it's related to networking or the database.

I added a time trigger to my Function that runs a simple database query every 3 mins to keep both the Function itself and the database in a "hot state" so they never fall asleep - because a cold start takes 35-75 secs in my case, which is unacceptable.

**Comment by danbasszeti** [comment]
I'm currently having this issue, mostly when I've cancelled requests that are taking ridiculously long due to the new .NET 8 isolated cold start and then retrying them

**Comment by tom08zehn** [comment]
In my case, the problem was solved by switching the app platform settings from 32 to 64 bit.

I had random HTTP 500 errors and the logs showed that it happened when a database connection was established. My guess is that the connection was broken at some point due to Azure's resource allocation, e.g. when cold-starting the database and sometimes also during runtime... maybe virtual memory reallocation?

**Comment by dantstore** [comment]
Just wanted to add my two cents here, for any other poor souls losing their mind over this issue. We have had, and continue to have intermittent issues with the same error message. However, we only see them during deployment with Bicep. I can only assume the cold start is to blame, but I have also heard mention that the function keys associated with a function app are generated separately, and are not included in the Bicep 'ready' check. I'm aware that this is not the exact same issue, but I'm hoping this might nudge someone else in the right direction

**Comment by isadag** [comment]
I've had multiple outages throughout the day. I have alerting via an external service that monitors my function app, and I am getting a http status code 503 back from the function host, not my own code..it is not reaching my code/function. 

Any updates on this? There is no workaround or way to get it up and running again and it is extremely frustrating

**Comment by isadag** [comment]
A follow up.. I created a new function app in the West Europe region and running my function app on Windows instead of Linux. My previous function app that is in Sweden Central on Linux has reported multiple outages from my external monitoring solution. The same function app in West Europe has had no outages yet. It's been up and running since deploy (~14 hours).


What's up with the function app platform? Is it a Linux issue? Sweden central issue? Azure's status page reports that everything is green....

**Comment by koberghe** [comment]
We also experienced intermittent 502 and 503 errors several times a day starting in mid-August. Despite trying various things, including upgrading from the v1 to the v2 programming model and changing regions, nothing resolved the issue. Then, suddenly in mid-September, these issues disappeared, and we haven't had a single outage in days.

**Comment by isadag** [comment]
> We also experienced intermittent 502 and 503 errors several times a day starting in mid-August. Despite trying various things, including upgrading from the v1 to the v2 programming model and changing regions, nothing resolved the issue. Then, suddenly in mid-September, these issues disappeared, and we haven't had a single outage in days.

I've migrated my function to a new one running on windows in Western Europe. I haven't experienced the amount of outages as I did when running on Linux in Sweden Central. I've seen a few alerts, but the service has been down just a few minutes at a time, and not nearly as often as before. But everytime they occur, the Azure status page shows nothing, everything seems to be up and running according to it

**Comment by koberghe** [comment]
> > We also experienced intermittent 502 and 503 errors several times a day starting in mid-August. Despite trying various things, including upgrading from the v1 to the v2 programming model and changing regions, nothing resolved the issue. Then, suddenly in mid-September, these issues disappeared, and we haven't had a single outage in days.
> 
> I've migrated my function to a new one running on windows in Western Europe. I haven't experienced the amount of outages as I did when running on Linux in Sweden Central. I've seen a few alerts, but the service has been down just a few minutes at a time, and not nearly as often as before. But everytime they occur, the Azure status page shows nothing, everything seems to be up and running according to it

I believe these downtimes were visible in "Diagnose and solve problems" -> "Web App Down"

**Comment by kshyju** [comment]
Following up on some older issues. The last update was over a year ago. The best way to get answers for your app specific issues is to open a support ticket so the team can look into your app. At this point. I don't see anything actionable here. So, marking this as closed. Feel free to open a new issue if you still experience any other issues with azure functions.

**Comment by isadag** [comment]
> Following up on some older issues. The last update was over a year ago. The best way to get answers for your app specific issues is to open a support ticket so the team can look into your app. At this point. I don't see anything actionable here. So, marking this as closed. Feel free to open a new issue if you still experience any other issues with azure functions.

Given that many of us here have reported similar issues while the status pages etc report no issues whatsoever during these periods of errors, I feel we are entitled to some form of answer or explanation as to what is going on with the services. As we've proved that it has nothing to do with our code, and have been able to get the same issues when deploying brand new sample function apps in the same region, I'd say the community here should get some form of support and reassurance that Microsoft actually will look into the matter.

We can share more info about time periods when these issues occur,  but for now I have had to pause the monitoring of some checks as they've been fluctuating so much and caused enormous amounts of alert notifications here.

**Comment by rdvansloten** [comment]
> > Following up on some older issues. The last update was over a year ago. The best way to get answers for your app specific issues is to open a support ticket so the team can look into your app. At this point. I don't see anything actionable here. So, marking this as closed. Feel free to open a new issue if you still experience any other issues with azure functions.
> 
> Given that many of us here have reported similar issues while the status pages etc report no issues whatsoever during these periods of errors, I feel we are entitled to some form of answer or explanation as to what is going on with the services. As we've proved that it has nothing to do with our code, and have been able to get the same issues when deploying brand new sample function apps in the same region, I'd say the community here should get some form of support and reassurance that Microsoft actually will look into the matter.
> 
> We can share more info about time periods when these issues occur,  but for now I have had to pause the monitoring of some checks as they've been fluctuating so much and caused enormous amounts of alert notifications here.

Unless you're a big enterprise buying isolated clusters, I don't think they give a shit. Azure Functions are easily one of their weakest offerings. I've moved off this service in favor of k8s  and spot ARM VMs and it's been a load off my mind.
